src/map/map_engine.h

```diff
@@ -33,32 +33,32 @@
 // Forward Declarations
 //
 
-class MapServer;
 class SqlConnection;
 class MapNetworking;
 class MapStatistics;
 class CZone;
 
+struct MapConfig final
+{
+    std::string ip{};
+    uint16      port{ 0 };
```

Comment:
You could store the two of these as an `IPP` so they get access to all the battle-tested conversion host/network bits